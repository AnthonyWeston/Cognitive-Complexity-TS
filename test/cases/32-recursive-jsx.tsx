function Component() {
    return <RecursiveSelfClosingComponent/>;
}

function RecursiveSelfClosingComponent() {
    return <RecursiveSelfClosingComponent/>;
}

function RecursiveComponent() {
    return <RecursiveComponent></RecursiveComponent>
}
