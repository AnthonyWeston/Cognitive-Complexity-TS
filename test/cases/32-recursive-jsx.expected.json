{
    "32-recursive-jsx.tsx": {
        "score": 2,
        "inner": [
            {
                "name": "Component",
                "score": 0,
                "line": 1,
                "column": 1
            },
            {
                "name": "RecursiveSelfClosingComponent",
                "score": 1,
                "line": 5,
                "column": 1
            },
            {
                "name": "RecursiveComponent",
                "score": 1,
                "line": 9,
                "column": 1
            }
        ]
    }
}